import "goby/core/proto/option_extensions.proto";

message TesMoosAppConfig
{
  // Global moos variables (Specified outside "ProcessConfig" blocks)
  optional bool log = 100 [default = true, (moos_global) = "log",
                           (description)="Should we write a text log of the terminal output?"];
  optional string log_path = 101 [default = "./",
                                  (moos_global) = "log_path",
                                  (description)="Directory path to write the text log of the terminal output (if log=true)"];
  optional string community = 102 [(moos_global) = "Community",
                                   (description)="The vehicle's name", (example)="AUV23"];
  optional double lat_origin = 103 [(moos_global) = "LatOrigin",
                                    (description)="Latitude in decimal degrees of the local cartesian datum",
                                    (example)="42.5"];
  optional double lon_origin = 104 [(moos_global) = "LongOrigin",
                                    (description)="Longitude in decimal degrees of the local cartesian datum",
                                    (example)="10.9"];
  
  optional uint32 app_tick = 1 [default = 10,
                                (description)="Frequency at which to run Iterate()."];
  
  optional uint32 comm_tick = 2 [default = 10,
                                 (description)="Frequency at which to call into the MOOSDB for mail."];
  
  enum Verbosity { VERBOSITY_QUIET = 1; VERBOSITY_WARN = 2; VERBOSITY_VERBOSE = 3; VERBOSITY_DEBUG = 4; VERBOSITY_GUI = 5; }
  optional Verbosity verbosity = 3 [default = VERBOSITY_VERBOSE,
                                    (description)="Verbosity of the terminal window output"];

  // post various configuration values to the MOOSDB on startup
  message Initializer
  {
    enum Type
    {
      INI_DOUBLE = 1;
      INI_STRING = 2;
    }
    required Type type = 1 [(description)="type of MOOS variable to publish",
                            (example)="INI_DOUBLE"];
    required string moos_var = 2 [(description)="name of MOOS variable to publish to",
                                  (example)="SOME_MOOS_VAR"];
    optional string global_cfg_var = 3 [(description)="Optionally, instead of giving `sval` or `dval`, give a name here of a global MOOS variable (one at the top of the file) whose contents should be written to `moos_var`", (example)="LatOrigin"]; // pull from globals at top of MOOS file
    optional double dval = 4 [(description)="Value to write for type==INI_DOUBLE",
                              (example)="3.454"]; // required if type == MOOS_DOUBLE
    optional string sval = 5 [(description)="Value to write for type==INI_STRING",
                              (example)="a string"]; // required if type == MOOS_STRING
  }  
  repeated Initializer initializer = 10 [(description)="Publish a constant value to the MOOSDB at startup"];
}
    
